### PR TITLE
Knn remove mappers

### PR DESCRIPTION
**Pull request by korshaknn**
## Description

The problem is that we map directions responses into our own types in the `MapboxOffboardRouter` and map them back to `mapbox-java's` `DirectionsResponse` when passing to the legacy UI code. 

Model classes from `mapbox-java` were moved to a separate module `mapbox-sdk-directions-models`, so we depend on it and don't have to convert models twice in our logic.

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

## Testing

Please describe the manual tests that you ran to verify your changes

- [x] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [x] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: libdirections-offboard/src/main/java/com/mapbox/navigation/route/offboard/MapboxOffboardRouter.kt [hunk 1]
```diff
@@ -20,24 +20,24 @@ class MapboxOffboardRouter(private val accessToken: String, private val context:
...
+        mapboxDirections?.enqueueCall(object : Callback<DirectionsResponse> {
 
             override fun onResponse(
                 call: Call<DirectionsResponse>,
                 response: Response<DirectionsResponse>
             ) {
                 val routes = response.body()?.routes()
                 if (response.isSuccessful && !routes.isNullOrEmpty()) {
-                    callback.onResponse(routes.map { it.mapToRoute() })
+                    callback.onResponse(routes.map { it })
```
Comment (LukasPaczos) [inline libdirections-offboard/src/main/java/com/mapbox/navigation/route/offboard/MapboxOffboardRouter.kt]: Unnecessary mapping :)

File: libdirections-offboard/src/main/java/com/mapbox/navigation/route/offboard/router/MapboxDirectionsUtils.kt [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+private val EVENT_LISTENER = NavigationRouteEventListener()
+private const val SEMICOLON = ";"
+private const val COMMA = ","
+
+fun MapboxDirections.Builder.routeOptions(options: RouteOptions): MapboxDirections.Builder {
+    baseUrl(options.baseUrl())
+    user(options.user())
+    profile(options.profile())
+
+    options.coordinates().forEachIndexed { index, point ->
```
Comment (LukasPaczos) [inline libdirections-offboard/src/main/java/com/mapbox/navigation/route/offboard/router/MapboxDirectionsUtils.kt]: We should assert that there are at least 2 points in the array.

File: libdirections-onboard/src/main/java/com/mapbox/navigation/route/onboard/MapboxOnboardRouter.kt [hunk 3]
```diff
@@ -58,26 +58,32 @@ class MapboxOnboardRouter : Router {
     }
 
     override fun getRoute(
-        routeOptions: RouteOptionsNavigation,
+        routeOptions: RouteOptions,
         callback: Router.Callback
     ) {
+
+        val origin = routeOptions.coordinates().first()
```
Comment (LukasPaczos) [inline libdirections-onboard/src/main/java/com/mapbox/navigation/route/onboard/MapboxOnboardRouter.kt]: As above, we should assert that there are at least 2 points and throw a meaningful error if needed.

File: libdirections-offboard/src/main/java/com/mapbox/navigation/route/offboard/router/MapboxDirectionsUtils.kt [hunk 4]
```diff
@@ -0,0 +1,226 @@
+package com.mapbox.navigation.route.offboard.router
```
Comment (LukasPaczos) [inline libdirections-offboard/src/main/java/com/mapbox/navigation/route/offboard/router/MapboxDirectionsUtils.kt]: Can we move this class to base and a add `@file:JvmName("MapboxDirectionsUtils")`?

Comment (korshaknn) [reply libdirections-offboard/src/main/java/com/mapbox/navigation/route/offboard/router/MapboxDirectionsUtils.kt]: there is an extension `MapboxDirections.Builder.routeOptions` . Base has no dependency on `MapboxDirections`. Right?

File: libdirections-offboard/src/main/java/com/mapbox/navigation/route/offboard/router/MapboxDirectionsUtils.kt (hunk 2 above)
Comment (korshaknn) [reply libdirections-offboard/src/main/java/com/mapbox/navigation/route/offboard/router/MapboxDirectionsUtils.kt]: done

File: libdirections-onboard/src/main/java/com/mapbox/navigation/route/onboard/MapboxOnboardRouter.kt (hunk 3 above)
Comment (korshaknn) [reply libdirections-onboard/src/main/java/com/mapbox/navigation/route/onboard/MapboxOnboardRouter.kt]: done

File: libdirections-offboard/src/main/java/com/mapbox/navigation/route/offboard/MapboxOffboardRouter.kt (hunk 1 above)
Comment (korshaknn) [reply libdirections-offboard/src/main/java/com/mapbox/navigation/route/offboard/MapboxOffboardRouter.kt]: removed

File: libdirections-offboard/src/main/java/com/mapbox/navigation/route/offboard/router/MapboxDirectionsUtils.kt (hunk 4 above)
Comment (LukasPaczos) [reply libdirections-offboard/src/main/java/com/mapbox/navigation/route/offboard/router/MapboxDirectionsUtils.kt]: Yup, good catch. Let's move that one extension to a different file and leave it in offbaord module and move the rest to base.

Comment (korshaknn) [reply libdirections-offboard/src/main/java/com/mapbox/navigation/route/offboard/router/MapboxDirectionsUtils.kt]: done

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/extensions/RouteOptionsUtils.kt [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+            .toTypedArray()
+    val waypoints = arrayOfNulls<Point>(splitWaypointTargets.size)
+    var index = 0
+    for (waypointTarget in splitWaypointTargets) {
+        val point = waypointTarget.split(COMMA.toRegex()).dropLastWhile { it.isEmpty() }
+            .toTypedArray()
+        if (waypointTarget.isEmpty()) {
+            waypoints[index++] = null
+        } else {
+            val longitude = java.lang.Double.valueOf(point[0])
```
Comment (Guardiola31337) [inline libnavigation-base/src/main/java/com/mapbox/navigation/base/extensions/RouteOptionsUtils.kt]:
```suggestion
            val longitude = Double.valueOf(point[0])
```

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/extensions/RouteOptionsUtils.kt [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+            val latitude = java.lang.Double.valueOf(point[0])
```
Comment (Guardiola31337) [inline libnavigation-base/src/main/java/com/mapbox/navigation/base/extensions/RouteOptionsUtils.kt]:
```suggestion
            val latitude = Double.valueOf(point[0])
```

File: libnavigation-base/src/main/java/com/mapbox/navigation/base/extensions/RouteOptionsUtils.kt (hunk 5 above)
Comment (korshaknn) [reply libnavigation-base/src/main/java/com/mapbox/navigation/base/extensions/RouteOptionsUtils.kt]: changed to `point[0].toDouble()`

**Pull request merged**